rustic-cargo.el

```diff
@@ -111,6 +111,47 @@ If ARG is not nil, use value as argument and store it in `rustic-test-arguments'
                               :process proc
                               :mode mode)))
 
+;;;###autoload
+(defun rustic-cargo-current-test ()
+  "Run 'cargo test' for the test near point."
+  (interactive)
+  (rustic-compilation-process-live)
+  (rustic-compilation-start (list rustic-cargo-bin "test" (rustic-cargo--get-current-fn-fullname))
+                            :buffer rustic-test-buffer-name
+                            :process rustic-test-process-name
+                            :mode 'rustic-cargo-test-mode))
+
+(defun rustic-cargo--get-current-fn-fullname()
+  "Return full name of the fn around point including module name if any."
+  (let ((mod (rustic-cargo--get-current-mod))
+        (fn (rustic-cargo--get-current-fn-name)))
+    (if mod
+        (concat mod "::" fn)
+      fn)))
+
+(defconst rustic-cargo-mod-regexp "^[[:space:]]*mod[[:space:]]+\\([[:word:][:multibyte:]_][[:word:][:multibyte:]_[:digit:]]*\\)[[:space:]]*{")
+(defconst rustic-cargo-fn-regexp "^[[:space:]]*fn[[:space:]]+\\([^(]+\\)[[:space:]]*(")
+
+(defun rustic-cargo--get-current-mod ()
+  "Return mod name around pount or nil."
+  (save-excursion
+    (when (search-backward-regexp rustic-cargo-mod-regexp nil t)
+      (match-string 1))))
+
+(defun rustic-cargo--get-current-line-fn-name()
+  "Return fn name from the current line or nil."
+  (save-excursion)
```

Comment:
The argument is missing.

Reply:
What do you mean?

Reply:
You didn't add the function body to the `save-excursion` call. This will throw an error.

Reply:
Ah, right, thanks ;) I wish it had type system around. Btw, have you seen [Elsa](https://github.com/emacs-elsa/Elsa)?

Reply:
Yeah, but I didn't try it. You can run `eval-buffer` to catch those errors. Or `byte-recompile-directory` should also work.

Reply:
Well, it's not an error from the elisp perspective, so those commands won't help in this case.

Reply:
Oh, you just helped me to find a bug in remacs. Thanks ;)